server/store/user_store.go

```diff
@@ -239,16 +253,16 @@ func (s *pluginStore) DeleteUserFromIndex(mattermostUserID string) error {
 	})
 }
 
-func (s *pluginStore) SearchInUserIndex(term string, limit int) ([]UserShort, error) {
+func (s *pluginStore) SearchInUserIndex(term string, limit int) (UserIndex, error) {
 	userIndex, err := s.LoadUserIndex()
 	if err != nil {
 		return nil, errors.Wrap(err, "error searching user in index")
 	}
 
-	result := make([]UserShort, 0)
+	result := make([]*UserShort, 0)
```

Comment:
I'm used to the syntax of the following, and I would prefer to stay consistent across the codebase (and other plugin codebases). Mainly a coding style reference though

```suggestion
	result := []*UserShort{}
```